proto/minder/v1/minder.proto

```diff
@@ -1784,7 +1823,72 @@ message ListProvidersResponse {
     string cursor = 2;
 }
 
-// ProviderType is the type of the provider.
+message CreateProviderRequest {
+    // context is the context in which the provider is created.
+    Context context = 1;
+    // provider is the provider to be created.
+    Provider provider = 2;
+}
+
+message CreateProviderResponse {
+    // provider is the provider that was created.
+    Provider provider = 1;
+
+    // authorization provides additional authorization information needed
+    // to complete the initialization of the provider.
+    AuthorizationParams authorization = 2;
```

Comment:
There's probably a way to tie this in with the `AuthorizationFlow` so the client knows what to do with the authrorization parameters https://github.com/stacklok/minder/blob/790a8e8ae5496913e6cd26a4515c27456dfba983/proto/minder/v1/minder.proto#L1821-L1826

Reply:
Yes, I think different `AuthorizationFlow`s would correspond to different `oneof` parameters in the `ProviderParameter` struct.